trunion/tests/test_receipts.py

```diff
@@ -0,0 +1,129 @@
+                                          iat=self.signing['exp'] + 1))
+        self.assertRaises(HTTPConflict, valid_receipt, request)
+
+    def test_validate_user(self):
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          user='not a dict!'))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          user={}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          user={'type': ''}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          user={'type': 'taco',
+                                                'value': ''}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          user={'type': 'email',
+                                                'value': ''}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          user={'type': 'email',
+                                                'value': 'hal@9000'}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+    def test_validate_product(self):
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          product='not a dict!'))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          product={}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        request = StupidRequest(path=self.path,
+                                post=dict(self._template,
+                                          product={'url': ''}))
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        post = dict(self._template,
+                    product={'url': 'gopher://yoyodyne-propulsion.com'})
+        request = StupidRequest(path=self.path, post=post)
+        self.assertRaises(HTTPBadRequest, valid_receipt, request)
+
+        post = dict(self._template,
+                    product={'url': 'https://grumpybadgers.com',
+                             'storedata': ''})
```

Comment:
any special reason for using dict(...) instead of just well, a dict? this one is dict(...) containing a dict {}...